ckanext/canada/logic.py

```diff
@@ -460,3 +464,67 @@ def canada_job_list(up_func, context, data_dict):
         job['status'] = job_obj.get_status()
 
     return job_list
+
+
+@chained_action
+def canada_user_update(up_func, context, data_dict):
+    """
+    Checks if the sysadmin value is being changed.
+
+    We only want sysadmins from TBS to be able to do this.
+
+    We have to do it in this chained action, and not auth method
+    because sysadmins skip auth checks.
+    """
+    user = model.User.get(get_or_bust(data_dict, 'id'))
+
+    if not user:
+        raise ObjectNotFound(_('User not found'))
+
+    if data_dict.get('sysadmin', user.sysadmin) != user.sysadmin:
```

Comment:
validation of the data_dict hasn't happened here yet, so if someone passes a string version of this parameter it will always show as not equal.

Reply:
@wardi I just copied whatever was in core: https://github.com/ckan/ckan/blob/master/ckan/logic/action/update.py#L797

Reply:
Not sure what you mean, I don't see where core is doing `data_dict.get('sysadmin', user.sysadmin) != user.sysadmin`. In core there isn't a place to hook in to the action after validation but before model_save. This could be a new validator in the update user schema instead.

Reply:
@wardi oops my bad, read that comment wrong. Okay I am now calling the validate and then checking `sysadmin` in the validated data